editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -2599,6 +2599,11 @@ void CanvasItemEditor::_gui_input_viewport(const Ref<InputEvent> &p_event) {
 
 void CanvasItemEditor::_update_cursor() {
 	CursorShape c = CURSOR_ARROW;
+	bool should_switch = false;
+	if (drag_selection.size() != 0) {
+		float angle = drag_selection[0]->_edit_get_rotation();
+		should_switch = abs(Vector2(1, 0).rotated(angle).dot(Vector2(1, 0))) < Math_SQRT12;
```

Comment:
Sorry for the many reviews telling you to change your code, but I thought of an even better solution:

```suggestion
		should_switch = abs(Math::cos(angle)) < Math_SQRT12;
```

Reply:
I'm usually pretty good at trigonometry, I don't know why I didn't think of that... Anyway, thanks. It makes sense and it's fast. Comitting.